components/Modal/CancelBidModal.js

```diff
@@ -0,0 +1,168 @@
+import useStore from 'lib/store'
+import { useIntl } from 'hooks/useIntl'
+import Modal from 'components/Common/Modal'
+import Button from 'components/Common/Button'
+import { sentryCaptureException } from 'lib/sentry'
+import { GAS_FEE, STORAGE_ADD_MARKET_FEE } from 'config/constants'
+import JSBI from 'jsbi'
+import { IconX } from 'components/Icons'
+import { useState } from 'react'
+import WalletHelper from 'lib/WalletHelper'
+import { trackUpdateListingToken } from 'lib/ga'
+import { useToast } from 'hooks/useToast'
+
+const CancelBidModal = ({ data, show, onClose, onSuccess, tokenType = `token` }) => {
+	const { localeLn } = useIntl()
+	const [isCancelBid, setIsCancelBid] = useState(false)
+	const { currentUser, setTransactionRes } = useStore((state) => ({
+		currentUser: state.currentUser,
+		setTransactionRes: state.setTransactionRes,
+	}))
+	const toast = useToast()
+
+	const hasStorageBalance = async () => {
+		try {
+			const currentStorage = await WalletHelper.viewFunction({
+				methodName: 'storage_balance_of',
+				contractId: process.env.MARKETPLACE_CONTRACT_ID,
+				args: { account_id: currentUser },
+			})
+
+			const supplyPerOwner = await WalletHelper.viewFunction({
+				methodName: 'get_supply_by_owner_id',
+				contractId: process.env.MARKETPLACE_CONTRACT_ID,
+				args: { account_id: currentUser },
+			})
+
+			const usedStorage = JSBI.multiply(
+				JSBI.BigInt(parseInt(supplyPerOwner) + 1),
+				JSBI.BigInt(STORAGE_ADD_MARKET_FEE)
+			)
+
+			if (JSBI.greaterThanOrEqual(JSBI.BigInt(currentStorage), usedStorage)) {
+				return true
+			}
+			return false
+		} catch (err) {
+			sentryCaptureException(err)
+		}
+	}
+
+	const onCancelBid = async () => {
+		setIsCancelBid(true)
+
+		trackUpdateListingToken(data.token_id)
+
+		const hasDepositStorage = await hasStorageBalance()
+
+		try {
+			const depositParams = { receiver_id: currentUser }
+
+			const params = {
+				nft_contract_id: data.contract_id,
+				...(data.token_id
+					? { token_id: data.token_id }
+					: { token_series_id: data.token_series_id }),
+				token_id: data.token_id,
+				account_id: currentUser,
+			}
```

Comment:
kalo gini ada masalah ga?
```suggestion
			const params = {
				nft_contract_id: data.contract_id,
				token_id: data.token_id,
				account_id: currentUser,
			}
```

Reply:
done hu